### PR TITLE
executor: fix panic when limit is too large

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

When limit is too large. `TopN.totalLimit` will overflow int and got panicked.

### What is changed and how it works?

Let `TopN.totalLimit` become uint64.

Let me know if you have better idea.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by zz-jason** [comment]
@winoros please add some labels.

File: executor/sort.go [hunk 1]
```diff
@@ -307,7 +307,7 @@ func (e *TopNExec) Next(ctx context.Context, chk *chunk.Chunk) error {
 	}
 	chk.Reset()
 	if !e.fetched {
-		e.totalLimit = int(e.limit.Offset + e.limit.Count)
+		e.totalLimit = e.limit.Offset + e.limit.Count
```
Comment (lamxTyler) [inline executor/sort.go]: Will it overflow uint64?

Comment (XuHuaiyu) [reply executor/sort.go]: Yes, but it seems to happen with little probability in a real scenario.

**Comment by shenli** [comment]
Good job!
We should check if there is any other similar case.

**Pull request merged**